src/coreclr/System.Private.CoreLib/src/System/Collections/Generic/ComparerHelpers.cs

```diff
@@ -103,11 +71,6 @@ internal static object CreateDefaultEqualityComparer(Type type)
                 // Specialize for byte so Array.IndexOf is faster.
                 result = new ByteEqualityComparer();
             }
-            else if (type == typeof(string))
-            {
-                // Specialize for string, as EqualityComparer<string>.Default is on the startup path
```

Comment:
Based on the comment, this looks like a startup time optimization. Does this introduce more work during startup?

Reply:
I was wondering if it maybe makes sense to return the specialized `StringComparer.Ordinal` instead to avoid using the shared generic for string, what do you think?

Reply:
I do not think that it is going to work. StringComparer is not assignable to EqualityComparer<string>.

It may be a good idea to replace EqualityComparer<string> with StringComparer.Ordinal if it helps us to avoid creating EqualityComparer<string> for hello world. (I am not sure whether it is possible.)

Reply:
> It may be a good idea to replace EqualityComparer with StringComparer.Ordinal if it helps us to avoid creating EqualityComparer for hello world. (I am not sure whether it is possible.)

The usage seems to be here: https://github.com/dotnet/runtime/blob/eaa9717d90115cea43b4cdd7a2a49e6d3c3d780e/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs#L67
It doesn't really seem possible to replace. I don't think R2R will understand the reflection path so I guess I should return the special-case?

Reply:
Yep.